File: bike-shop/bike-shop/frontend/src/services/postService.js
import axiosClient from "./axiosConfig";

const postService = {
  getPublished: () => axiosClient.get("/posts"),
  getById: (id) => axiosClient.get(`/posts/${id}`),
  getBySlug: (slug) => axiosClient.get(`/posts/slug/${slug}`),
  getAllAdmin: () => axiosClient.get("/posts/admin/all"),
  create: (data) => axiosClient.post("/posts", data),
  update: (id, data) => axiosClient.put(`/posts/${id}`, data),
  toggleVisibility: (id) => axiosClient.patch(`/posts/${id}/toggle-visibility`),
  delete: (id) => axiosClient.delete(`/posts/${id}`),
};

export default postService;

File: bike-shop/bike-shop/backend/src/main/java/com/example/bike_shop/service/PostService.java
package com.example.bike_shop.service;

import com.example.bike_shop.dto.PostDTO;

import java.util.List;

public interface PostService {
    List<PostDTO> getPublished();
    List<PostDTO> getAll();
    PostDTO getById(Long id);
    PostDTO getBySlug(String slug);
    PostDTO create(PostDTO dto);
    PostDTO update(Long id, PostDTO dto);
    PostDTO toggleVisibility(Long id);
    void delete(Long id);
}
